mystring.cpp

```diff
@@ -71,4 +71,40 @@ ostream & operator<< (ostream& os, const String &s)
 unsigned int String::length() const
 {  
     return size;
+}
+
+//overloading assigment operator
+String & String::operator=(const String &obj)
+{
+    //guard self assignment
+    if(this != &obj)
+    {
+        size = obj.size;
+        strcpy(text.get(), obj.text.get());
+    }
+
+    return *this;
+}
+
+//function which insert substring in specified position
+const char *String::insert(const String &str, size_t pos)
+{
+    const char *result = nullptr;
+
+    //check if given position is inside object string
+    if((pos < (this->length())) && (pos > 0))
```

Comment:
you don't need to write "this" every time you address some class members -- it's already presumed you're there. the only case where it it might be useful is when you have 2 variables with similar name. but you have named your class members according to some pattern to distinguish them, didn't you? ;-)

Reply:
ok, "this" removed